base/pkg/read.jl

```diff
@@ -3,18 +3,29 @@
 module Read
 
 import ...LibGit2, ..Cache, ..Reqs, ...Pkg.PkgError, ..Dir
-using ..Types
+using ..Types, Base.Filesystem
 
-readstrip(path...) = strip(readstring(joinpath(path...)))
+isgitrepo(pkg) =
+    ispath_casesensitive(joinpath(pkg, ".git"))
 
-url(pkg::AbstractString) = readstrip(Dir.path("METADATA"), pkg, "url")
-sha1(pkg::AbstractString, ver::VersionNumber) = readstrip(Dir.path("METADATA"), pkg, "versions", string(ver), "sha1")
+readstrip(path...) =
+    strip(readstring(joinpath(path...)))
+
+url(pkg::AbstractString) =
+    readstrip(Dir.path("METADATA"), pkg, "url")
+
+sha1(pkg::AbstractString, ver::VersionNumber) =
+    readstrip(Dir.path("METADATA"), pkg, "versions", string(ver), "sha1")
 
 function available(names=readdir("METADATA"))
     pkgs = Dict{String,Dict{VersionNumber,Available}}()
     for pkg in names
-        isfile("METADATA", pkg, "url") || continue
-        versdir = joinpath("METADATA", pkg, "versions")
+        pkgdir = joinpath("METADATA", pkg)
+        if !isdir_casesensitive(pkgdir)
```

Comment:
Is this really necessary?   In cases where `pkg` is supplied by the METADATA, not by the user, can't we assume that the case is correct?